aptos-move/aptos-gas/src/gas_meter.rs

```diff
@@ -162,7 +162,7 @@ impl AptosGasMeter {
     ) -> Self {
         assert!(
             (gas_feature_version == 0 && storage_gas_params.is_none())
-                || storage_gas_params.is_some(),
+                || (gas_feature_version > 0 && storage_gas_params.is_some()),
```

Comment:
Is there a unit test? If not, can we add one?

Reply:
It would be hard. This should be unreachable if we code the client (AptosVM) correctly...

Reply:
Or do you mean having a simple unit test that tries to call this constructor with some value combinations? Not sure if this brings much value though...

Reply:
How about e2e tests?

Reply:
Not possible. It is by design that the AptosVM never calls this constructor with the invalid combinations.